cores/esp32/HardwareSerial.cpp

```diff
@@ -261,19 +261,25 @@ void HardwareSerial::begin(unsigned long baud, uint32_t config, int8_t rxPin, in
     if (!uartIsDriverInstalled(_uart)) {
         switch (_uart_nr) {
             case UART_NUM_0:
-                rxPin = rxPin < 0 ? SOC_RX0 : rxPin;
-                txPin = txPin < 0 ? SOC_TX0 : txPin;
+                if (rxPin < 0 && txPin < 0) {
```

Comment:
This doesn't make sense to me.  The previous code 
```
rxPin = rxPin < 0 ? SOC_RX0 : rxPin;
txPin = txPin < 0 ? SOC_TX0 : txPin;
```
allowed changing pins individually.  With this code if rxPin is set then txPin won't be set to default.

Reply:
@mrengineer7777  See https://github.com/espressif/arduino-esp32/discussions/6356#discussioncomment-2311636

Reply:
Yes, exactly. Previous code was forcing any undefined pin to go to the default IO Pin, only in the first ```Serial.begin()``` call.

Now it only forces default pins when no pin is defined.

When only defining rxPin, tx IOMUX will be unset. 
Thus, ```Serial.write()``` would not have any effect, but it also doesn't crash ESP32. 
The same for txPin alone -> ```Serial.read()``` will always return -1.

Reply:
> This doesn't make sense to me. The previous code
> allowed changing pins individually. With this code if rxPin is set then txPin won't be set to default.

Please note that this code only applies to the first time Serial.begin() is called.
Subsequent calls of Serial.begin() will change rxPin and/or TxPin individually.

For example:
```cpp
Serial1.begin(9600, SERIAL_8N1, 12, 14);   // first time calling begin will set rxPin and txPin
Serial1.begin(115200);                     // doesn't change any pin (rx/tx), but only baudrate
Serial1.begin(115200, SERIAL_8N1, -1, 27); // only changes txPin to 27, keeping rxPin in 12
Serial1.end();                             // no seeting is kept from the setup above
Serial1.begin(115200);                     // new begin() from an end() will configure rxPin and txPin 
                                           // to DEFAULT pins (Serial1 9 and 10 on ESP32)
```

Reply:
Ok, so this is a deliberate rollback to previous behavior.  Looks like the crash was being caused by the TX pin defaulting to a pin that's not available on the user's hardware.  One question: does passing TX=-1 to the underlying driver uartBegin() cause any issues?  I'm guessing not based on this comment:
>>Thus, Serial.write() would not have any effect, but it also doesn't crash ESP32.

Reply:
No issues with TX or RX as -1. ```uartBegin()``` will just not associate its digital pad to any IO_MUX signal.
The consequence is that the function related to the pin won't work (sending, receiving, etc). 
Anyway, this feature must be used with caution.